packages/core-mobile/app/hooks/watchlist/useGetPrices.ts

```diff
@@ -7,13 +7,14 @@ import { selectSelectedCurrency } from 'store/settings/currency'
 import { Prices } from 'store/watchlist'
 
 export const useGetPrices = (
-  tokenIds: string[]
+  tokenIds: string[],
+  shouldGetPrices = true
 ): UseQueryResult<Prices, Error> => {
   const currency = useSelector(selectSelectedCurrency).toLowerCase()
   const isFocused = useIsFocused()
 
   return useQuery({
-    enabled: isFocused && tokenIds.length > 0,
+    enabled: isFocused && tokenIds.length > 0 && shouldGetPrices,
```

Comment:
I think you can just pass in `enabled` and let this logic lives in the component